magniv/build/build.py

```diff
@@ -40,64 +40,74 @@ def get_tasks_from_file(filepath, root, req, used_keys) -> Dict:
+                info = {
+                    "location": filepath,
+                    "name": node.name,
+                    "python_version": _get_python_version(root),
+                    "owner": _get_owner(root),
+                    "requirements_location": req,
+                    "line_number": node.lineno,
+                    "key": node.name,
+                    "description": None,
+                }
+                for kw in decorator.keywords:
+                    info[kw.arg] = kw.value.value
+                missing_reqs = list({"schedule"} - set(info))
+                if len(missing_reqs) > 0:
+                    raise ValueError(
+                        "Task missing required variables, please resolve by defining ("
+                        + ",".join([f" {x} " for x in missing_reqs])
+                        + ") in the task decorator"
+                    )
+                if info["key"] in used_keys:
+                    raise ValueError(
+                        f'Task "{info["key"]}" in file {filepath} is using "{info["key"]}" as a key which is already used in {used_keys[info["key"]]}, please resolve by changing one of the keys'
+                    )
+                else:
+                    used_keys[info["key"]] = filepath
+                tasks.append(info)
+    return tasks, used_keys
 
 
-def build():
+def save_tasks(
+    task_folder: str,
+    dump_save_pth: str = "./dump.json",
+    reqs_pth: str = "/requirements.txt",
+    root_req: None = None,
+    tasks_list: List = [],
+    used_keys: Dict = {},
+) -> NoReturn:
     """
-    We walk through the `tasks` directory, and for each file we find, we check if it's a Python file,
-    and if it is, we get the tasks from it and add them to our list of tasks
+    Save the decorated tasks to a json file.
+    
+    :param task_folder: The folder where the tasks are located
+    :param dump_save_pth: The path to the file where the dump will be saved, defaults to ./dump.json
+    :param reqs_pth: The path to the requirements.txt file, defaults to /requirements.txt
+    :param root_req: The root requirement file to use, defaults to None
+    :param tasks_list: The list of tasks to save, defaults to []
+    :param used_keys: The dictionary of used keys, defaults to {}
+    :return: NoReturn
     """
-    tasks_list = []
-    used_keys = {}
-    root_req = None
-    if not os.path.exists("./tasks"):
-        raise OSError("You must have a tasks folder that contains all of your tasks files")
-
-    # Hack to fix project imports by adding project directory to Python path
-    sys.path.append(os.getcwd() + "/tasks")
-
-    for root, dirs, files in os.walk("./tasks"):
-        if os.path.exists("./tasks/requirements.txt"):
-            root_req = "./tasks/requirements.txt"
+    for root, dirs, files in os.walk(task_folder):
+        if os.path.exists(task_folder + reqs_pth):
+            root_req = task_folder + reqs_pth
         req = (
```

Comment:
@seanabreau I ran this on a local tasks folder where i have tasks nested in folders with a requirements.txt in one of the nested folders. 
It didnt use that nested requirments.txt as it should have. I think this line might be an issue or something
![Screen Shot 2022-06-16 at 6 20 59 PM](https://user-images.githubusercontent.com/5013466/174187149-c84e0166-df17-4e51-8efe-de62009496fa.png)

Reply:
@Jped good catch give it a shot now